src/offset_date_time.rs

```diff
@@ -546,6 +546,110 @@ impl OffsetDateTime {
     }
 }
 
+/// Methods that replace the following components `Date`, `Time`, `PrimitiveDateTime`, `Offset`.
+impl OffsetDateTime {
+    /// Replacing with a new `Time` component. The new `PrimitiveDateTime` value is stored as UTC internally.
+    ///
+    /// The return value is a `Offset_date_time`.
+    ///
+    /// ```rust
+    ///   # use time_macros::{datetime,offset,time};
+    /// assert_eq!(
+    ///     datetime!("2020-01-01 5:00")
+    ///         .assume_offset(offset!("UTC"))
+    ///         .replace_time(time!("12:00")),
+    ///     datetime!("2020-01-01 12:00").assume_offset(offset!("UTC"))
+    /// );
+    /// assert_eq!(
+    ///     datetime!("2020-01-01 12:00")
+    ///         .assume_offset(offset!("-5"))
+    ///         .replace_time(time!("7:00")),
+    ///     datetime!("2020-01-01 12:00").assume_offset(offset!("UTC"))
```

Comment:
It would be best to avoid changing the offset between the left and right side, as it makes it less clear that the behavior is correct.